src/util/parented_ptr.h

```diff
@@ -58,22 +58,6 @@ class parented_ptr {
         return m_pObject;
     }
 
-    bool operator== (const parented_ptr& other) const {
-        return m_pObject == other.m_pObject;
-    }
-
-    bool operator== (const T* other) const {
-        return m_pObject == other;
-    }
-
-    bool operator!= (const parented_ptr& other) const {
```

Comment:
operator != should always be implemented in terms of operator ==:
return !(lhs == rhs);